### PR TITLE
Reprompt if a non-existent public directory is used in firebase init command.

### PR DESCRIPTION
**Pull request by TamalSaha**
fyi: @jwngr

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ var defaultSettings = {
 
 function getPrompt(argv, schema, onComplete, index, results) {
   if (!Array.isArray(schema)) {
-    console.log(chalk.red('An error occurred'));
+    console.log(chalk.red('getPrompt schema must be an Array.'));
```
Comment (drtriumph) [inline lib/app.js]: This isn't useful to the average developer running `firebase-tools`, in fact it's pretty confusing. This repo needs a lot of work on messaging, but I don't think this is at all helpful here

File: lib/app.js [hunk 2]
```diff
@@ -404,17 +405,20 @@ module.exports = {
...
           }
         }
-      }];
-
-      getPrompt(argv, schema, function(results) {
-        if (path.relative('.', results.public).match(/^\./)) {
-          console.log(chalk.red('init cancelled - the public directory must be within the current working directory'));
-          process.exit(1);
+      };
+      schemaPublic.conform = function(value) {
```
Comment (drtriumph) [inline lib/app.js]: there's no reason this couldn't be defined inline like `schemaPublic.before()`

File: lib/app.js (hunk 1 above)
Comment (TamalSaha) [inline lib/app.js]: My understanding is that it is a debug message. Our users can never see this even for invalid inputs. So, I just changed that to something we can use for debugging. Do you want me to revert it ?

File: lib/app.js [hunk 3]
```diff
@@ -404,17 +405,20 @@ module.exports = {
...
+        if (path.relative('.', value).match(/^\./)) {
+          schemaPublic.message = 'Please use a sub directory of current working directory';
```
Comment (drtriumph) [inline lib/app.js]: Perhaps something like:

'Please ensure the specified public directory is within the current working directory'

File: lib/app.js [hunk 4]
```diff
@@ -404,17 +405,20 @@ module.exports = {
...
+          return false;
         }
-        if (!fs.existsSync(results.public)) {
-          console.log(chalk.red('init cancelled - the directory you entered does not exist'));
-          process.exit(1);
+        if (!fs.existsSync(value)) {
+          schemaPublic.message = 'Please use an existing directory';
```
Comment (drtriumph) [inline lib/app.js]: 'The specified public directory must exist'

File: lib/app.js (hunk 1 above)
Comment (drtriumph) [inline lib/app.js]: Ok, that's a fair point. Lets leave it as what you have here - we need a complete messaging overhaul soon

File: lib/app.js (hunk 3 above)
Comment (drtriumph) [inline lib/app.js]: reasoning: I don't like 'sub' on its own (it should be hyphenated at least) but the terminology is perhaps a little technical. People understand directories in other directories better than 'sub-directories' I believe

File: lib/app.js (hunk 4 above)
Comment (drtriumph) [inline lib/app.js]: reasoning: 'Please use an existing directory' doesn't have a clear CTA and implies that the folder has to already exist. There's nothing to stop someone creating a folder in the interim and running the prompt again. I actually wonder if we should create the folder for them here, or at least only check this when actually deploying to Firebase Hosting (they might want to have a build folder that is generated by some script later)

**Comment by drtriumph** [comment]
A few comments, back to you. Given the lack of tests anyway, ignore the coveralls failure above

File: lib/app.js (hunk 3 above)
Comment (TamalSaha) [inline lib/app.js]: Done.

File: lib/app.js (hunk 2 above)
Comment (TamalSaha) [inline lib/app.js]: Done.

File: lib/app.js (hunk 4 above)
Comment (TamalSaha) [inline lib/app.js]: "I actually wonder if we should create the folder for them here, or at least only check this when actually deploying to Firebase Hosting (they might want to have a build folder that is generated by some script later)" - I don't know. The current restrictions look reasonable to me.

Comment (drtriumph) [inline lib/app.js]: The current restrictions work, but perhaps could be relaxed to facilitate some more advanced workflows. Lets leave it as is for now, but consider changing this in a future release

File: lib/app.js [hunk 5]
```diff
@@ -403,18 +404,21 @@ module.exports = {
           else {
             return value;
           }
+        },
+        conform: function(value) {
+          if (path.relative('.', value).match(/^\./)) {
+            schemaPublic.message = 'Please ensure the specified public directory is within the current working directory';
```
Comment (drtriumph) [inline lib/app.js]: Please use a directory within the current working directory

File: lib/app.js [hunk 6]
```diff
@@ -403,18 +404,21 @@ module.exports = {
...
+            return false;
+          }
+          if (!fs.existsSync(value)) {
+            schemaPublic.message = 'The specified public directory must exist';
```
Comment (drtriumph) [inline lib/app.js]: The specified directory must exist

**Pull request merged**